src/sparseml/pytorch/sparsification/quantization/helpers.py

```diff
@@ -712,16 +712,16 @@ def _delete_get_block_hooks(
         post_hooks = []
 
         # get first and last Module objects in block by their names
-        block_head = get_layer(block[0], module)
-        block_tail = get_layer(block[-1], module)
+        for name in block:
+            m = get_layer(name, module)
 
-        for handle_id, pre_hook_fn in list(block_head._forward_pre_hooks.items()):
-            pre_hooks.append(pre_hook_fn)
-            del block_head._forward_pre_hooks[handle_id]
+            for handle_id, pre_hook_fn in list(m._forward_pre_hooks.items()):
```

Comment:
Wonder if you'd mind reduce the duplicate pattern from these two for loops (make it easier to read)?

Reply:
Updated - is the new logic easier to read?

Reply:
Absolute. Thanks!